includes/class-stats.php

```diff
@@ -256,23 +257,43 @@ public function get_order_earnings( $query = array() ) {
 		) );
 
 		if ( true === $this->query_vars['relative'] ) {
+
+			$initial_query = "SELECT IFNULL({$function}, 0) AS total
```

Comment:
I'm finding this a lot easier to read and have two questions:
1. Should the `$initial_query` match the default `else` query? I'm seeing differences (`IFNULL` and the else doesn't have a `where_sql`), but if they should be the same, maybe we can tinker with the logic to just set up that query once and then add the relative as needed.
2. I can't comment on line 250 directly, but I'm seeing `edd_subscription` in the array of statuses twice. Maybe we can run that line through `array_unique` after/outside of `apply_filters`?

Reply:
That would hypothetically change this entire block beginning at line 259 to:

```php
		$initial_query = "SELECT IFNULL({$function}, 0) AS total
			FROM {$this->query_vars['table']}
			WHERE 1=1
			{$this->query_vars['status_sql']}
			{$this->query_vars['type_sql']}
			{$this->query_vars['currency_sql']}
			{$this->query_vars['where_sql']}
			{$this->query_vars['date_query_sql']}";

		$initial_result = $this->get_db()->get_row( $initial_query );

		if ( true === $this->query_vars['relative'] ) {

			$relative_date_query_sql = $this->generate_relative_date_query_sql();

			$relative_query = "SELECT IFNULL({$function}, 0) AS total
				FROM {$this->query_vars['table']}
				WHERE 1=1
				{$this->query_vars['status_sql']}
				{$this->query_vars['type_sql']}
				{$this->query_vars['currency_sql']}
				{$this->query_vars['where_sql']}
				{$relative_date_query_sql}";

			$relative_result = $this->get_db()->get_row( $relative_query );
		}

		$total = null === $initial_result->total
			? 0.00
			: (float) $initial_result->total;

		if ( true === $this->query_vars['relative'] ) {
			$total = $this->generate_relative_markup( floatval( $total ), floatval( $relative_result->total ) );
		} else {
			$total = $this->maybe_format( $total );
		}
```

But that's assuming that the `IFNULL` and `where_query` are acceptable in both queries (non-relative as well as relative).